lib/tpm-v2.c

```diff
@@ -399,6 +399,23 @@ u32 tpm2_get_capability(struct udevice *dev, u32 capability, u32 property,
 	return 0;
 }
 
+u32 tpm2_algo_get_mask_from_hash(enum tpm2_algorithms hash)
```

Comment:
You are confusing two things here.

TCG2_BOOT_HASH_ALG_SHA256 etc are the mask that the EFI TCG protocol expects. 
What we return now is the active PCR banks that the TPM spec expects. So I think this patch is wrong apart from the renaming parrt